src/compile/data/source.ts

```diff
@@ -65,13 +66,13 @@ export class SourceNode extends DataFlowNode {
     this._name = name;
   }
 
-  set parent(parent: DataFlowNode) {
-    throw new Error('Source nodes have to be roots.');
-  }
+  // set parent(parent: DataFlowNode) {
```

Comment:
yeah. because of ordering I was trying to properly set the parent. I think this was a mistake, but it was what I sort of understood to this point. In my head lookup Source Nodes != root Source Nodes.... which I guess is wrong? Was having difficult reasoning through that.